Tweaking trumail logic

lguegan opened the pull request:
Don't throw 500's when we are blocked by the receiver, just mark deliverable as unknown

lguegan committed:
Tweaking trumail logic

lguegan committed:
Add module file

nathan-pr commented on api/lookup.go:
```diff
@@ -28,16 +29,22 @@ func LookupHandler(v *verifier.Verifier) echo.HandlerFunc {
+		deliverable := "true"
 		if err != nil {
-			return FormatEncoder(c, http.StatusInternalServerError, err)
+			if strings.Contains(err.Error(), "Fatal") {
```
If your'e prefixing with Fatal, would a string.startsWith be faster if there's such a thing in go?

nathan-pr reviewed (approved):
One question, but other than that, seems reasonable

lguegan committed:
Fixing weird go issue

lguegan committed:
Only make timeout fatal

nithinsubhash committed:
Corrected module name

Corrected module name to module github.com/fullcontact/trumail to support `go get github.com/fullcontact/trumail` installs.

nithinsubhash committed:
Updated error handling logic

- Added a flag to LookupError to mark it as fatal
- Added a field `message` to the Lookup struct. The error string is added to this field if the error is not fatal

nithinsubhash committed:
Updated echo version, added prometheus exporter

nithinsubhash committed:
Reduced MX dial timeout

Reduced MX server dial timeout to 50s to prevent ALB idle timeout of 60s from happening.

nithinsubhash committed:
Reduced timeout to 40s

nithinsubhash committed:
Increased timeout to 45s

nithinsubhash committed:
Added disposable emails txt

nithinsubhash committed:
Added disposable flag, reformatted code

nithinsubhash committed:
Use tag instead of branchname

nithinsubhash committed:
Removed some disposable email domains

nithinsubhash committed:
Version update

nithinsubhash committed:
Removed an ignored error

nithinsubhash committed:
Added error details in the response

nithinsubhash committed:
Fix to handle SPF errors separately

nithinsubhash committed:
Merge pull request #2 from fullcontact/updated-error-handling

Updated error handling logic
